src/test/java/com/telefonica/iot/cygnus/sinks/OrionMongoSinkTest.java

```diff
@@ -0,0 +1,185 @@
+ * General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ * fiware-cygnus is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the
+ * implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with fiware-cygnus. If not, see
+ * http://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License please contact with iot_support at tid dot es
+ */
+package com.telefonica.iot.cygnus.sinks;
+
+import com.telefonica.iot.cygnus.backends.mongo.MongoBackend;
+import com.telefonica.iot.cygnus.containers.NotifyContextRequest;
+import com.telefonica.iot.cygnus.sinks.OrionMongoSink.DataModel;
+import com.telefonica.iot.cygnus.utils.Constants;
+import com.telefonica.iot.cygnus.utils.TestUtils;
+import java.util.HashMap;
+import org.apache.flume.Context;
+import org.apache.flume.channel.MemoryChannel;
+import org.apache.flume.lifecycle.LifecycleState;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import static org.mockito.Mockito.doNothing;
+import org.mockito.runners.MockitoJUnitRunner;
+
+/**
+ *
+ * @author frb
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class OrionMongoSinkTest {
+    
+    @Mock
+    private MongoBackend mockMongoBackend;
+    
+    // instance to be tested
+    private OrionMongoSink sink;
+    
+    // other instances
+    private Context context;
+    private NotifyContextRequest notifyContextRequest;
+    
+    // constants
+    private final String mongoHost = "localhost";
+    private final String mongoPort = "27017";
+    private final String mongoUsername = "admin";
+    private final String mongoPassword = "1a2b3c4d";
+    private final String dataModel = "collection-per-entity";
+    private final String dbPrefix = "test_";
+    private final String collectionPrefix = "test_";
+    private final String dbName = "db-name";
+    private final String collectionName = "collection-name";
+    private final long recvTimeTs = 1429535775;
+    private final String recvTime = "2015-04-20T12:13:22.41";
+    private final String entityId = "car1";
+    private final String entityType = "car";
+    private final String attrName = "speed";
+    private final String attrType = "kmh";
+    private final String attrValue = "112.9";
+    private final String attrMd = "[]";
+    private final String serviceHeader = "vehicles";
+    private final String servicePathHeader = "4wheels";
+    private final String destination = "4wheelsCars";
+    private final String notifyXMLSimple = ""
```

Comment:
Regarding the test, it does not matter if the notification is XML or Json written, since it is parsed as a `NotifyContextRequest` object, which is really used in the end.